### PR TITLE
Alerting: edit receivers (grafana)

### PR DESCRIPTION
**Pull request by domasx2**
* Create/update grafana managed receivers 
* **Cloud alertmanager receiver editing not implemented yet** coming with next PR
* Testing the receivers is not in scope of this PR as well

- [x] figure out why posting hangs
- [x] test with backend, especially validation error handling
- [x] handle the case where existing receiver was renamed (need to update routes)

~~There's a bug with how config resturned by API: https://raintank-corp.slack.com/archives/C01LJ5F8NRX/p1619425374104500
But faik the UI is correct~~ fixed

https://user-images.githubusercontent.com/847684/116058018-14396180-a688-11eb-9d00-520b5915539c.mp4

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/features/alerting/unified/RuleEditor.tsx [hunk 1]
```diff
@@ -45,9 +46,7 @@ const ExistingRuleEditor: FC<ExistingRuleEditorProps> = ({ identifier }) => {
       <Page.Contents>
         <InfoBox severity="warning" title="Rule not found">
           <p>Sorry! This rule does not exist.</p>
-          <a href="/alerting/list">
-            <Button>To rule list</Button>
-          </a>
+          <LinkButton href={locationUtil.assureBaseUrl('/alerting/list')}>To rule list</LinkButton>
```
Comment (torkelo) [inline public/app/features/alerting/unified/RuleEditor.tsx:49]: do not think locationUtil.assureBaseUrl is needed

Comment (domasx2) [reply public/app/features/alerting/unified/RuleEditor.tsx]: removed

File: public/app/features/alerting/unified/components/receivers/ReceiversTable.tsx [hunk 2]
```diff
@@ -63,10 +63,10 @@ export const ReceiversTable: FC<Props> = ({ config, alertManagerName }) => {
                     `/alerting/notifications/receivers/${encodeURIComponent(receiver.name)}/edit`,
                     alertManagerName
                   )}
-                  tooltip="edit receiver"
+                  tooltip="edit contact point"
```
Comment (torkelo) [inline public/app/features/alerting/unified/components/receivers/ReceiversTable.tsx:66]:
```suggestion
                  tooltip="Edit contact point"
```

File: public/app/features/alerting/unified/components/receivers/ReceiversTable.tsx [hunk 3]
```diff
@@ -63,10 +63,10 @@ export const ReceiversTable: FC<Props> = ({ config, alertManagerName }) => {
...
                   icon="pen"
                 />
-                <ActionIcon tooltip="delete receiver" icon="trash-alt" />
+                <ActionIcon tooltip="delete contact point" icon="trash-alt" />
```
Comment (torkelo) [inline public/app/features/alerting/unified/components/receivers/ReceiversTable.tsx:69]:
```suggestion
                <ActionIcon tooltip="Delete contact point" icon="trash-alt" />
```

File: public/app/features/alerting/unified/components/receivers/form/ReceiverForm.tsx [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+const getStyles = (theme: GrafanaThemeV2) => ({
+  heading: css`
+    margin: ${theme.spacing(4, 0)};
+  `,
+  buttons: css`
+    margin-top: ${theme.spacing(4)};
+
+    & > * + * {
+      margin-left: ${theme.spacing(1)};
+    }
```
Comment (torkelo) [inline public/app/features/alerting/unified/components/receivers/form/ReceiverForm.tsx:131]: what is this for?

**Review by torkelo — CHANGES_REQUESTED** [review]
looks good!  but a lot of code and no unit tests. Some basic coverage would be good, and receiver-form.ts functions could especially need some.

File: public/app/features/alerting/unified/components/receivers/form/ReceiverForm.tsx (hunk 4 above)
Comment (domasx2) [reply public/app/features/alerting/unified/components/receivers/form/ReceiverForm.tsx:131]: adds spacing between child nodes (buttons), so they're not glued together

**Comment by domasx2** [comment]
> looks good! but a lot of code and no unit tests. Some basic coverage would be good, and receiver-form.ts functions could especially need some.

@torkelo You're right, but we are **extremely** pressed for time to get MVP in place for 8.0. One of the obvious corners to cut is automated test coverage. Working with the understanding that we'll be able to clean up all the tech debt once all the features are in, post release.

File: public/app/features/alerting/unified/components/receivers/form/ReceiverForm.tsx (hunk 4 above)
Comment (torkelo) [reply public/app/features/alerting/unified/components/receivers/form/ReceiverForm.tsx:131]: Can you use a HorizonalGroup instead?

Comment (domasx2) [reply public/app/features/alerting/unified/components/receivers/form/ReceiverForm.tsx:131]: hah, did not know this exists. Thanks, will try
- [ ] use HorizontalGroup

**Review by torkelo — APPROVED** [review]
@domasx2 I understand, it's just that with unit tests that rarely ever happens (ie go back and write unit tests to old code) unless you have a story that requires changes and you have a way of working that requires some level of unit tests. But to just write unit tests,  in my experience, there are always more important things to do than just writing unit tests for old code. 

I can understand waiting for extensive automated e2e tests until things are more in place and finalized. But some very basic level of unit tests should be part of the normal way of working. Hope the new alerting backend is not written without unit tests and a hope that it is something to be done "later", which usually results in hard-to-test & poorly structured code.  Luckily with UI component tests that is not as true (but is true for non-component logic). 

I am not asking for 100% test coverage, but some super basic react-testing "do these pages render" without exceptions tests, and some for the code in receiver-form, going to be more work to do later when your not as familiar with the requirements.

**Comment by domasx2** [comment]
@torkelo  OK, will add some tests for this soon, when I get the cloud alertmanager version of the form working. 

>  it's just that with unit tests that rarely ever happens (ie go back and write unit tests to old code)

As a sort-of-positive example, it did happen with the original Cloud Alerting UI plugin! We added tech debt cleanup as a goal and key objective for the next quarter, and followed through on it post release. Ofc I'd much rather not be in a rush to begin with

**Pull request merged**